pandas/tests/arithmetic/test_datetime64.py

```diff
@@ -2436,3 +2436,34 @@ def test_dt64arr_addsub_object_dtype_2d():
 
     assert result2.shape == (4, 1)
     assert all(td._value == 0 for td in result2.ravel())
+
+
+def test_concat_float_datetime64():
+    # GH#32934
+    df_time = pd.DataFrame({"A": pd.array(["2000"], dtype="datetime64[ns]")})
+    df_float = pd.DataFrame({"A": pd.array([1.0], dtype="float64")})
+
+    tm.assert_equal(
```

Comment:
For each of these:

1. Could you use `tm.assert_frame_equal`
2. Create variables `expected = ... , result = ...`